bacap UnmarshalBinary panics with nil; no way for external package to create an empty instance

if a go struct contains a pointer to bacap.UniversalReadCap and is Unmarshaled, bacap.UniversalReadCap.UnmarshalBinary points at a nil rootPublicKey, but there is no way to initialize these fields from another package.

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6da208]

goroutine 38 [running]:
testing.tRunner.func1.2({0xbb6640, 0xf41d30})
	/usr/lib/go/src/testing/testing.go:1632 +0x3fc
testing.tRunner.func1()
	/usr/lib/go/src/testing/testing.go:1635 +0x6b6
panic({0xbb6640?, 0xf41d30?})
	/usr/lib/go/src/runtime/panic.go:791 +0x132
github.com/katzenpost/hpqc/sign/ed25519.(*PublicKey).FromBytes(0x0, {0xc00047b235, 0x20, 0xbbafc0?})
	/root/go/pkg/mod/github.com/katzenpost/hpqc@v0.0.54/sign/ed25519/eddsa.go:277 +0x68
github.com/katzenpost/hpqc/bacap.(*UniversalReadCap).UnmarshalBinary(0xc000422930, {0xc00047b235, 0x88, 0x5bd})
	/root/go/pkg/mod/github.com/katzenpost/hpqc@v0.0.54/bacap/bacap.go:464 +0xf1
github.com/fxamacker/cbor/v2.fillByteString(0x40, {0xc00047b235, 0x88, 0x5bd}, 0x1, {0xbd52a0?, 0xc000422930?, 0xc0004608f0?}, 0x0, 0x0)
```

--- a/bacap/bacap.go
+++ b/bacap/bacap.go
@@ -461,6 +461,7 @@ func (u *UniversalReadCap) UnmarshalBinary(data []byte) error {
 	if len(data) != UniversalReadCapSize {
 		return errors.New("invalid UniversalReadCap binary size")
 	}
+	u.rootPublicKey = new(ed25519.PublicKey)
 	err := u.rootPublicKey.FromBytes(data[:32])
 	if err != nil {
 		return err

--- a/bacap/bacap.go
+++ b/bacap/bacap.go
@@ -398,6 +398,7 @@ func (o *BoxOwnerCap) UnmarshalBinary(data []byte) error {
 	if len(data) != BoxOwnerCapSize {
 		return errors.New("invalid BoxOwnerCap binary size")
 	}
+	o.rootPrivateKey = new(ed25519.PrivateKey)
 	err := o.rootPrivateKey.FromBytes(data[:64])
 	if err != nil {
 		return err

